### PR TITLE
[Debug Mode] Show reason for RBR/GBR on LHN tabs

### PR DESCRIPTION
**Pull request by pac-guerreiro**
### Fixed Issues

$https://github.com/Expensify/App/issues/49600
PROPOSAL: N/A

### Tests

<details>

<summary>To trigger a GBR in Inbox</summary>

**Make sure there are no reports with RBR**
1. Open any report from `Inbox` page
2. Click on the `+` on the chat editor and select `Assign Task`
3. Fill all the necessary data and assign the task to your self
4. The inbox should now show a green indicator and a message saying 'There's a report awaiting action' above the bottom bar
5. Open another report, then click on `View` button after the message 
6. You should be redirected to the report where you assigned the task to yourself

</details>

<details>

<summary>To trigger a RBR in Inbox</summary>

1. Create a new expense
2. Remove a required property from expense (e.g. `Category`)
3. The inbox indicator should be red
4. A message saying 'There's a report with errors' should be shown above the bottom bar
5. If you click on `View`, you should be redirected to the expense report

</details>

<details>

<summary>To trigger a GBR in Settings</summary>

1. Go to `Profile`, then open `Contact method`
2. Add new contact method
3. The new contact method should show a GBR indicator, so as settings avatar in the bottom bar
4. A message saying 'A contact method requires verification' should be displayed above the bottom bar
5. If you click on `View`, you should be redirected to the contact method page

</details>

<details>

<summary>To trigger a RBR in Settings</summary>

1. Go to `Profile`, then open `Contact method`
6. Add new contact method
8. During verification phase enter any code, then continue
9. The new contact method should show a RBR indicator, so as settings avatar in the bottom bar
10. A message saying 'There's a problem with a contact method' should be displayed above the bottom bar
11. If you click on `View`, you should be redirected to the contact method page

</details>

### Offline tests

Same as tests

### QA Steps

Same as tests

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/8e7b0508-89ce-4fa0-a579-920559a43604

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/595c6cb2-b3ba-4bc5-b3e6-d21009f5182a

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/d36362a9-de8e-459a-a320-7403c9b2a702

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/54c9473c-75c1-4437-a087-d25f4c142761

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/00437c6d-8130-44f6-9496-86ebf5e36e4e

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/afcd5d5d-68e5-4914-b7bf-8bb85bb16a5a

</details>

### CODE REVIEW

**Comment by pac-guerreiro** [comment]
I need to add unit tests for the new hook that I created - `useSettingsStatus` - to make sure that it's working property since I extracted it's logic from `Indicator.tsx`, so it could be reused in debug mode.

Besides this, I also have to:

- Finish english translations
- Translate english to spanish
- Make `useSettingsStatus` return and optional `policyID` which is necessary to redirect user's to the policy pages that contain errors or require attention
- Test if the translations fit in the UI
- Take screenshots and add them to the PR
- Add testing steps to the PR

**Comment by pac-guerreiro** [comment]
Today's update:

- [x] Finish english translations
- [x] Make useSettingsStatus return and optional policyID which is necessary to redirect user's to the policy pages that contain errors or require attention

To continue tomorrow:
- [ ] Translate english to spanish
- [ ] Add unit tests for `useSettingStatus` hook
- [ ] Test if the translations fit in the UI
- [ ] Take screenshots and add them to the PR
- [ ] Add testing steps to the PR

File: src/libs/WorkspacesSettingsUtils.ts
Comment (fabioh8010) [inline src/libs/WorkspacesSettingsUtils.ts:141]: What happened to the previous logic? Are we maintaining it somewhere ?

File: src/libs/Navigation/AppNavigator/createCustomBottomTabNavigator/DebugTabView.tsx
Comment (fabioh8010) [inline src/libs/Navigation/AppNavigator/createCustomBottomTabNavigator/DebugTabView.tsx:155]:
```suggestion
                {message && <Text style={[styles.textWhite, styles.lh20]}>{translate(message)}</Text>}
```

File: src/CONST.ts
Comment (fabioh8010) [inline src/CONST.ts:5779]: I think this should be called `INDICATOR_STATUS`

File: src/hooks/useSettingsStatus.ts
Comment (fabioh8010) [inline src/hooks/useSettingsStatus.ts:18]:
```suggestion
type IndicatorStatus = {
    indicatorColor: string;
    status: ValueOf<typeof CONST.SETTINGS_STATUS> | undefined;
    idOfPolicyWithErrors: string | undefined;
};

function useIndicatorStatus(): IndicatorStatus {
```

Correct if I'm wrong, but shouldn't this be called `useIndicatorStatus` since its main purpose is to be used in the `Indicator` component?

If yes I think we should also rename `SETTINGS_STATUS` to `INDICATOR_STATUS`

File: src/languages/en.ts
Comment (fabioh8010) [inline src/languages/en.ts:4902]:
```suggestion
        indicatorStatus: {
```

Probably should be `indicatorStatus`

**Review by fabioh8010 — COMMENTED** [review]
Partial review

**Comment by pac-guerreiro** [comment]
Today's update:
- [x] Add unit tests for `useSettingStatus` hook
- [x] Addressed @fabioh8010's feedback

To continue tomorrow:
- [ ] Translate english to spanish
- [ ] Test if the translations fit in the UI
- [ ] Take screenshots and add them to the PR
- [ ] Add testing steps to the PR

**Comment by pac-guerreiro** [comment]
Today's update:
- [x] Improved english translations
- [x] Fixed wrong routes being used for contact method error or info
- [x] Fixed GBR/RBR message not fitting in the UI

Todo:
- [ ] Test subscription GBR/RBR, reimbursement account RBR, workspace employee list RBR and workspace custom unit RBR
- [ ] Translate english to spanish
- [ ] Take screenshots and add them to the PR
- [ ] Add testing steps to the PR

File: src/hooks/useIndicatorStatus.ts
Comment (fabioh8010) [inline src/hooks/useIndicatorStatus.ts:48]:
```suggestion
    const errorChecking: Partial<Record<IndicatorStatus, boolean>> = {
```

File: src/hooks/useIndicatorStatus.ts
Comment (fabioh8010) [inline src/hooks/useIndicatorStatus.ts:59]:
```suggestion
    const infoChecking: Partial<Record<IndicatorStatus, boolean>> = {
```

File: src/hooks/useIndicatorStatus.ts
Comment (fabioh8010) [inline src/hooks/useIndicatorStatus.ts:18]:
```suggestion
type IndicatorStatus = ValueOf<typeof CONST.INDICATOR_STATUS>;

type IndicatorStatusResult = {
    indicatorColor: string;
    status: IndicatorStatus | undefined;
    policyIDWithErrors: string | undefined;
};

function useIndicatorStatus(): IndicatorStatusResult {
```

So you can reuse `IndicatorStatus` below and in other files

File: src/hooks/useIndicatorStatus.ts
Comment (fabioh8010) [inline src/hooks/useIndicatorStatus.ts:67]:
```suggestion
    const status = (error ?? info) as IndicatorStatus | undefined;
```

File: src/libs/Navigation/AppNavigator/createCustomBottomTabNavigator/DebugTabView.tsx
Comment (fabioh8010) [inline src/libs/Navigation/AppNavigator/createCustomBottomTabNavigator/DebugTabView.tsx:28]:
```suggestion
function getSettingsMessage(status: IndicatorStatus | undefined): TranslationPaths | undefined {
```

File: src/libs/Navigation/AppNavigator/createCustomBottomTabNavigator/DebugTabView.tsx
Comment (fabioh8010) [inline src/libs/Navigation/AppNavigator/createCustomBottomTabNavigator/DebugTabView.tsx:59]:
```suggestion
function getSettingsRoute(status: IndicatorStatus | undefined, policyIDWithErrors = ''): Route | undefined {
```

File: src/libs/WorkspacesSettingsUtils.ts
Comment (fabioh8010) [inline src/libs/WorkspacesSettingsUtils.ts:123]:
```suggestion
function getChatTabBrickRoadReport(policyID?: string): OnyxEntry<Report> {
```

File: src/libs/WorkspacesSettingsUtils.ts
Comment (fabioh8010) [inline src/libs/WorkspacesSettingsUtils.ts:135]:
```suggestion
        if (!brickRoad) {
            return false;
        }
        return [CONST.BRICK_ROAD_INDICATOR_STATUS.INFO, CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR].includes(brickRoad);
```

File: tests/unit/useIndicatorStatusTest.tsx
Comment (fabioh8010) [inline tests/unit/useIndicatorStatusTest.tsx:12]:
```suggestion
const getMockForStatus = (status: IndicatorStatus) =>
```

File: src/libs/WorkspacesSettingsUtils.ts
Comment (pac-guerreiro) [reply src/libs/WorkspacesSettingsUtils.ts]: @fabioh8010 I got confused with my own code hehehe but I managed to restore the previous logic!

Thanks for this, great catch 😄

**Comment by pac-guerreiro** [comment]
I forgot to leave my update yesterday, here it is:
- [x] Test subscription GBR/RBR, reimbursement account RBR, workspace employee list RBR and workspace custom unit RBR
- [x] Improve english translations
- [x] Translate english to spanish
- [x] Apply @fabioh8010's feedback

Todo:
- [ ] Take screenshots and add them to the PR
- [ ] Add testing steps to the PR

**Comment by pac-guerreiro** [comment]
All tasks done. This PR is now ready for review!

**Review by fabioh8010 — COMMENTED** [review]
Minor comments

**Comment by pac-guerreiro** [comment]
Still waiting on final review from @DylanDylann

**Comment by pac-guerreiro** [comment]
Still waiting on final review from @DylanDylann

**Comment by DylanDylann** [comment]
https://github.com/user-attachments/assets/d8e85696-c3e4-47ec-a2f4-2e3d9d54cc1d

Let's use text color to be suitable with the background color

File: src/languages/es.ts
Comment (DylanDylann) [inline src/languages/es.ts:5575]: @pac-guerreiro NAB: Did you confirm these translations?

File: src/libs/Navigation/AppNavigator/createCustomBottomTabNavigator/DebugTabView.tsx
Comment (DylanDylann) [inline src/libs/Navigation/AppNavigator/createCustomBottomTabNavigator/DebugTabView.tsx:66]:
```suggestion
            return ROUTES.WORKSPACE_DISTANCE_RATES.getRoute(policyIDWithErrors);
```

**Comment by pac-guerreiro** [comment]
@DylanDylann all feedback addressed!

I'll let you know when someone confirms the spanish translations 😄

**Comment by DylanDylann** [comment]
@pac-guerreiro We have conflicts, please help to resolve it

**Comment by DylanDylann** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

File: src/libs/Navigation/AppNavigator/createCustomBottomTabNavigator/BottomTabBar.tsx
Comment (DylanDylann) [inline src/libs/Navigation/AppNavigator/createCustomBottomTabNavigator/BottomTabBar.tsx:116]: @pac-guerreiro  Only display this if users enable the debug feature

**Comment by pac-guerreiro** [comment]
@DylanDylann I just resolved the conflict and applied a fix for the problem you encountered 😄

**Review by DylanDylann — APPROVED** [review]

**Comment by pac-guerreiro** [comment]
Thank you @DylanDylann for the review! 😄

The PR is ready to be merged!

**Review by puneetlath — APPROVED** [review]
This looks great and very thorough. Thank you!

**Comment by puneetlath** [comment]
Hmm, all the tests are passing.

**Pull request merged**
